docs/design/contextual-tabs.md

```diff
@@ -0,0 +1,392 @@
+
+```javascript
+Office.onReady(async () => {
+    const contextualTabJSON = ' ... ' // Assign the JSON string.
+    const contextualTab = JSON.parse(contextualTabJSON);
+    await Office.ribbon.requestCreateControls(contextualTab);
+
+    await Excel.run(context => {
+        var charts = context.workbook.worksheets
+            .getActiveWorksheet()
+            .charts;
+        charts.onActivated.add(showDataTab);
+        charts.onDeactivated.add(hideDataTab);
+        return context.sync();
+    });
+});
+```
+
+Next, define the handlers. The following is a simple example of a `showDataTab`, but see [Error Handling](#error-handling) below for a more robust version of the function. About this code, note:
+
+- Office controls when it updates the state of the ribbon. The  [Office.ribbon.requestUpdate](/javascript/api/office/office.ribbon?view=common-js&preserve-view=true#requestupdate-input-) method queues a request to update. The method will resolve the Promise object as soon as it has queued the request, not when the ribbon actually updates.
+- The parameter for the **requestUpdate** method is a [RibbonUpdaterData](/javascript/api/office/office.ribbonupdaterdata) object that (1) specifies the tab by its ID *exactly as specified in the JSON*; and (2) specifies visibility of the tab.
+- If you have more than one custom contextual tab that should be visible in the same context, you simply add additional tab objects to the `tabs` array.
+
+```javascript
+async function showDataTab() {
+    await Office.ribbon.requestUpdate({
+        tabs: [
+            {
+                id: "CtxTab1",
+                visibility: true
+            }
+        ]});
+}
+```
+
+The handler to hide the tab is nearly identical, except that it sets the `visiblity` property back to `false`.
+
+The Office JavaScript library also provides several interfaces (types) to make it easier to construct the **RibbonUpdateData** object. The following is the `showDataTab` function in TypeScript and it makes use of these types.
+
+```typescript
+const showDataTab = async () => {
+    const myContextualTab: Tab = {id: "CtxTab1", visibility: true};
+    const ribbonUpdater: RibbonUpdaterData = { tabs: [ myContextualTab ]};
+    await Office.ribbon.requestUpdate(ribbonUpdater);
+}
+```
+
+### Toggle tab visibility and the enabled status of a button at the same time
+
+The **requestUpdate** method is also used to toggle the enabled or disabled status of a custom button on either a custom contextual tab or a custom regular (non-contextual) tab. For details about this, see [Enable and Disable Add-in Commands](disable-add-in-commands.md). There may be scenarios in which you want to change both the visibility of a tab and the enabled status of a button at the same time. You can do this with a single call of **requestUpdate**. The following is an example in which a button on a non-contextual tab is enabled at the same time as a contextual tab is made visible.
```

Comment:
```suggestion
The **requestUpdate** method is also used to toggle the enabled or disabled status of a custom button on either a custom contextual tab or a custom regular (non-contextual) tab. For details about this, see [Enable and Disable Add-in Commands](disable-add-in-commands.md). There may be scenarios in which you want to change both the visibility of a tab and the enabled status of a button at the same time. You can do this with a single call of **requestUpdate**. The following is an example in which a button on a non-contextual tab is enabled at the same time a contextual tab is made visible:
```

Reply:
I think the "as" has to be there. 
Compare "As well as jailing him for eight weeks magistrates imposed another driving ban, which runs out at the same time as his current disqualification."